swatch-contracts/src/main/java/com/redhat/swatch/contract/resource/ContractsTestingResource.java

```diff
@@ -321,26 +321,39 @@ public RpcResponse syncAllSubscriptions(Boolean forceSync) throws ProcessingExce
   @RolesAllowed({"test", "support", "service"})
   public OfferingResponse syncOffering(String sku) throws ProcessingException {
     var response = new OfferingResponse();
+    SyncResult result = null;
     try {
       log.info("Sync for offering {}", sku);
-      SyncResult result = offeringSyncService.syncOffering(sku);
+      result = offeringSyncService.syncOffering(sku);
       response.detail(String.format("%s for offeringSku=\"%s\".", result, sku));
     } catch (Exception e) {
-      log.error("Error syncing offering", e.getMessage());
-      if (e.getCause() instanceof ApiException) {
-        ApiException apiException = (ApiException) e.getCause();
+      if (e.getCause() instanceof ApiException apiException) {
         switch (apiException.getResponse().getStatus()) {
           case 400:
+            log.error("Error syncing offering", e.getMessage());
             throw new BadRequestException(apiException.getMessage());
           case 403:
+            log.warn("Warning syncing offering", e.getMessage());
             throw new ForbiddenException(apiException.getMessage());
           case 404:
+            log.warn("Warning syncing offering", e.getMessage());
             throw new NotFoundException(apiException.getMessage());
           default:
+            log.error("Error syncing offering", e.getMessage());
```

Comment:
I don't think the new log traces added here add any value since we're throwing the exceptions with the actual information we need.